Py3 compat

jro1234 opened the pull request:
@marscher @franknoe @jhprinz @euhruska 
I have made small changes throughout the package. Many of them fix small errors or add some exception handling. While running scaling tests, a few filesystem or database-related errors occurred so these have been wrapped in try-except blocks. Most of them happened with only a few hundred workers, so it did not require very large numbers of workers to get the errors. In the openmmrun.py I have the input files read in shuffled order in addition to the exception handling. I also included some edits I made early on to the first tutorials, however most are left as-is for now. Will keep working on those. 

All of the current doctests and the 2 integration tests pass. I've had some delay from the integration tests, which seem to behave differently. While the main call and test setup methods each are written identically as far as I can tell, the 'wrapped' test runs in the root directory of Travis environment unless the directory is changed within the test code.

The most substantial change is to provide better a better featurizer interface to the MDFeaturizer. This happens through a dict containing MDFeaturizer attribute names, which can be used to create a feature selection in the modeller task. Original syntax was limited to something like {"add_selection": {"select_Backbone": None}} --> MDFeaturizer.add_selection(select_Backbone()). This dict can now be used to pass positional and kw args to the add_x method, and/or the inner select_x method call. This will probably be essential for practical use to provide custom atom selection strings read by MDTraj.

thempel commented on adaptivemd/condition.py:
```diff
@@ -76,6 +76,9 @@ def __invert__(self):
     def __bool__(self):
         return self()
 
+    def __nonzero__(self):
```
Could someone briefly explain what this function does and why it is needed here?

jro1234 replied:
This is for compatibility between python 2 and python 3. In 2, `bool(x)` will look for a method `x.__nonzero__` , and x.`__bool__` in 3. I have both there to evaluate properly in all cases. I thought it was nice to have `__nonzero__` call `__bool__` , so upon the eventual deprecation of python 2 this will just work as-is.

franknoe replied:
Please document all functions in the code. Do you know which doc style we are using? Have a look at PyEMMA.

thempel commented on adaptivemd/engine/openmm/openmmrun.py:
```diff
@@ -33,6 +33,88 @@
+             'get_integrator': [get_xml, integrator]}
+
+    kfuncs = list(funcs.keys())
+    random.shuffle(kfuncs)
```
Why??

jro1234 replied:
When running a large number of workers who start tasks simultaneously (100s to 1000s), there will be filesystem errors. On the Titan lustre filesystem, the rate of errors was maybe 0.1 to 0.5%. This function shuffles the file read order, desychronizing to help alleviate this a bit.

thempel commented:
Changes in engine/openmm/openmm.py:
If I get this correct, the changes introduce some kind of error handling for failed openmm runs. First, I don't understand why a second try should work if a first one fails. Normally, if openmm fails, something is wrong with the script. Second, this approach seems very prone to overwriting ones data (correct me if I'm wrong). Wouldn't it be better to fix the problems with openMM, if any?

What is the motivation for all these retrying-loops, e.g. in openmm/openmmrun.py or in worker.py? It seems to make things more complicated than before...

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

Merging RP database.py

vivek-bala committed:
adding db, resource manager and task manager classes and other other helper methods

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
database class will add 'storage' implicitly, tasks dict is brought to root level, begin work to separate config and resource

vivek-bala committed:
adding tmgr class

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
reverted having to move the '_dict' object to the root and added configurations to the resource

vivek-bala committed:
chpt

itomaldonado committed:
added client.close() to get_tasks, modified update_task to update state in both root and _dict, cleaned up imports

itomaldonado committed:
some class cleanup

itomaldonado committed:
resource config field change, configuration added to resources will not include the name

itomaldonado committed:
reverted the ability to merge configuration with resrouce, configurations are now returned as a list

vivek-bala committed:
local chpt

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
added a generic matcher and a configuration matcher util

vivek-bala committed:
wrapping up tmgr

itomaldonado committed:
fixed a problem where the update would not properly both update state entries

vivek-bala committed:
local chpt -- todo work on matchmaking between configuration and resource

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

vivek-bala committed:
first test pass

vivek-bala committed:
resource + configs verified at client

itomaldonado committed:
formatting updates

jro1234 committed:
Task-Resource-Configuration structure, unclean

vivek-bala committed:
verified translation of resource, configuration -- task translation pending output staging

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
added get file location and added unittests for the database

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

vivek-bala committed:
wrapping up utils tests

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

vivek-bala committed:
added method to get source files from generator collection

vivek-bala committed:
adding method to

vivek-bala committed:
local chpt

itomaldonado committed:
added hex_to_id utility function

itomaldonado committed:
added a way to get files from the generator collection

vivek-bala committed:
local chpt

itomaldonado committed:
added some docs

vivek-bala committed:
pushing all changes + fix setup file

itomaldonado committed:
removed the L from hex_uuid

itomaldonado committed:
merge upstream

vivek-bala committed:
taking out print statements

franknoe commented:
great 


Sent from my T-Mobile 4G LTE Device

-------- Original message --------
From: jrossyra <notifications@github.com> 
Date: 8/28/17  1:28 AM  (GMT+01:00) 
To: markovmodel/adaptivemd <adaptivemd@noreply.github.com> 
Cc: Frank Noe <frank.noe@fu-berlin.de>, Mention <mention@noreply.github.com> 
Subject: Re: [markovmodel/ada 

@jrossyra commented on this pull request.



In adaptivemd/engine/openmm/openmmrun.py:
> +                raise e
+
+def read_input(platform, pdbfile, system, integrator):
+    import time
+    import random
+
+    return_order = ['get_platform', 'get_pdbfile',
+                    'get_system', 'get_integrator']
+
+    funcs = {'get_platform':   [get_platform, platform],
+             'get_pdbfile':    [get_pdbfile, pdbfile],
+             'get_system':     [get_xml, system],
+             'get_integrator': [get_xml, integrator]}
+
+    kfuncs = list(funcs.keys())
+    random.shuffle(kfuncs)

When running a large number of workers who start tasks simultaneously (100s to 1000s), there will be filesystem errors. On the Titan lustre filesystem, the rate of errors was maybe 0.1 to 0.5%. This function shuffles the file read order, desychronizing to help alleviate this a bit.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub, or mute the thread.


  
  




{"api_version":"1.0","publisher":{"api_key":"05dde50f1d1a384dd78767c55493e4bb","name":"GitHub"},"entity":{"external_key":"github/markovmodel/adaptivemd","title":"markovmodel/adaptivemd","subtitle":"GitHub repository","main_image_url":"https://cloud.githubusercontent.com/assets/143418/17495839/a5054eac-5d88-11e6-95fc-7290892c7bb5.png","avatar_image_url":"https://cloud.githubusercontent.com/assets/143418/15842166/7c72db34-2c0b-11e6-9aed-b52498112777.png","action":{"name":"Open in GitHub","url":"https://github.com/markovmodel/adaptivemd"}},"updates":{"snippets":[{"icon":"PERSON","message":"@jrossyra commented on #56"}],"action":{"name":"View Pull Request","url":"https://github.com/markovmodel/adaptivemd/pull/56#discussion_r135424158"}}}

jro1234 commented:
w.r.t. to all these changes that definitely complicate the operations done by openmmrun.py...
When pushing the number of simultaneous workers into the range of 100s or more, there are a variety of errors that happen quite reliably. I have added all this, which is pretty specific to the openmm engine. The structure can be recycled for other engines too. There is also an extra wrapper around the execution line of openmm, because when calling so many instances of the python package simultaneously the package read/ import also can fail.

vivek-bala committed:
local chkpt

itomaldonado committed:
upgraded pymongo version

vivek-bala committed:
commit for testing with admd workload

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

jro1234 committed:
Take a list of conditions

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

vivek-bala committed:
local chpt -- need to get raw path of the input files in order to proceed

thempel commented:
Ok, thanks for the clarifications! Could you add some quick comments, especially regarding what we discussed here? Otherwise, future developers might just "simplify" your code again...

vivek-bala committed:
fixing setup.py

vivek-bala committed:
revert to older format and putting simple key rename

vivek-bala committed:
adding some more commands for cleanup and compatible

itomaldonado committed:
added a way to get the shared files from the generator

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
now the shared_files

jro1234 committed:
RP tests moved

vivek-bala committed:
local chpt

vivek-bala committed:
local chppt

jro1234 committed:
[configuration] documentation

vivek-bala committed:
checkpoint on first set of MD tasks

jro1234 committed:
notebook updates

jro1234 committed:
after meld

jro1234 committed:
notebook again

jro1234 committed:
notebook again 2

jro1234 committed:
resource name in PythonTask

jro1234 committed:
rollback queues to list

itomaldonado committed:
added a way to mark a file as 'created' with proper timestamping

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
resource name in PythonTask parent

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

vivek-bala committed:
local chpt

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

jro1234 committed:
added CRT properties to PythonTask

vivek-bala committed:
saving local

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

jro1234 committed:
more CRT

vivek-bala committed:
saving changes to notebook

vivek-bala committed:
fix conflict

jro1234 committed:
fixed super call, dirty

jro1234 committed:
no changes

jro1234 committed:
trace PythonTask insert failure

vivek-bala committed:
fixing conflict

jro1234 committed:
fixed resource_name check

vivek-bala committed:
fixing more and more and more conflicts

vivek-bala committed:
Merge branch 'rp_integration' of github.com:jrossyra/adaptivemd into rp_integration

itomaldonado committed:
parse modeller task commend properly

itomaldonado committed:
moved to use the in operator

itomaldonado committed:
added a util to generate the PythonTask input file

vivek-bala committed:
changes before demo

jro1234 committed:
no changes

itomaldonado committed:
added tests for complex pythontask_input utility function

Commit:
added travis build

jro1234 committed:
condition upgraded

jro1234 committed:
[configuration] upgrade, doesnt play well with Bundle

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
added missing set_store for configurations

jro1234 committed:
[Configuration] set_current akin to old 'Resource' for adaptivemdworkers

jro1234 committed:
Remove retry loop in openmmrun line

jro1234 committed:
Openmmsetup utility: change ES cutoff & timesteps

jro1234 committed:
Specify system filenames for openmmrun.py

jro1234 committed:
Changed queueing args and shared path expansion

jro1234 committed:
Small changes, arg for always random sampling

jro1234 committed:
Update __init__.py

jro1234 committed:
Update __init__.py

vivek-bala committed:
pending changes

vivek-bala committed:
adding notebook

vivek-bala committed:
Merge branch 'rp_integration' into installation_cleanup

vivek-bala committed:
fixing conflict

vivek-bala committed:
fix pymongo version

jro1234 committed:
[openmmrun] time, random import to top

vivek-bala committed:
changing pymongo version

vivek-bala committed:
switching installation order

jro1234 committed:
Update project.py

jro1234 committed:
Update mongodb.py

jro1234 committed:
temp fixes to export in task main

jro1234 committed:
gpu_per_node enabled for configuration

jro1234 committed:
[gitignore] *ipynb

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
[mongodb] fixed 1-deep circular load call (RP disabled)

jro1234 committed:
[mongodb] dictify build syntax adjusted

jro1234 committed:
undo syntax change, causes error

jro1234 committed:
small changes

jro1234 committed:
[task] provided env prepended to task descriptions

jro1234 committed:
[task] environment prefix added

jro1234 committed:
Deactivate environment if added

jro1234 committed:
[config] parser moved to utils

jro1234 committed:
retry added back

jro1234 committed:
Small changes, extra printouts

jro1234 committed:
Update setup.yaml

jro1234 committed:
Update __init__.py

jro1234 committed:
Added virtualenv to tasks

jro1234 committed:
Fixed typo

itomaldonado committed:
updated pilot description from walltime to runtime

itomaldonado committed:
better error handling when creating resource

itomaldonado committed:
typo in string formatting

jro1234 committed:
Fixed Exception kwarg error

jro1234 committed:
Trajectory Tasks inherit PrePostTask, add venv to prepost

itomaldonado committed:
some better task-to-cud generation

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

itomaldonado committed:
fixing the PythonTask

itomaldonado committed:
added actual CPU calculation and updated utils test

jro1234 committed:
Resource requirements passed from engine to task

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
Fix rp.utils typo, resource_name type

jro1234 committed:
Fixed passing of resource requirements

jro1234 committed:
Fixed resource requirements typo

itomaldonado committed:
typo in resource requirements

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

itomaldonado committed:
updated tests for utils, about 80% converage

Commit:
kinetic_map True

jro1234 committed:
Added Titan ORTE RP config

jro1234 committed:
Update configuration.py

jro1234 committed:
Resolve and fix slashes in path in rp.utils

itomaldonado committed:
more changes to test utils

itomaldonado committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

Commit:
chignolin added setup

Commit:
chignolin added setup fixed

itomaldonado committed:
if the task is not on list, dont udpate it

itomaldonado committed:
fixed issue with file_created and assumption about the move being the last item

jro1234 committed:
mdconvert to pretask, changes to file (need to undo)

itomaldonado committed:
normalize worker://

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

itomaldonado committed:
handle restart.npz

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

itomaldonado committed:
reversed staging output directive

itomaldonado committed:
added ability to handle TrajectoryExtensionTasks

jro1234 committed:
[engine] Argument handling

jro1234 committed:
Weird merge fixing

jro1234 committed:
logging replace prints

jro1234 committed:
failed pilot note

jro1234 committed:
Faster StoredBundle pick method

jro1234 committed:
[ObjectStore] save pathway added with insert_many capability

jro1234 committed:
[StoredBundle] handle group or item add

jro1234 committed:
[Project] queue list results in add list to bundle

jro1234 committed:
Small Fixes

jro1234 committed:
[Bundle] listadd

jro1234 committed:
[Bundle] list comprehension in add

jro1234 committed:
[Bundle] fixed list comprehension in add

jro1234 committed:
[bulk_taskqueue] esolve merge conflicts

jro1234 committed:
[cb_buffer] process implemented for callbacks

jro1234 committed:
Fix lost changes to bundle,configuration

jro1234 committed:
Callbacks buffered and processed in new proc

jro1234 committed:
Updated lost fixes

jro1234 committed:
CU IDs to adaptivemd task entry

jro1234 committed:
Reformat cb entries, added 'pending' state on cud creation and 'running' state in cb

jro1234 committed:
Merge branch 'callback_buffering' into rp_integration

jro1234 committed:
Small Changes

jro1234 committed:
Check scheduling of compute units

jro1234 committed:
working on gpus

jro1234 committed:
env loads first in pretask

jro1234 committed:
more robust model retrieval

jro1234 committed:
CUD compute pars

jro1234 committed:
NTL9 in Charmm22star after short equilibration

jro1234 committed:
Chignolin in Charmm22star after short equilibration

jro1234 committed:
Merge branch 'cancel_cus' into rp_integration

jro1234 committed:
CUD mapping updates

jro1234 committed:
Merge branch 'cancel_cus' of https://github.com/jrossyra/adaptivemd into cancel_cus

jro1234 committed:
Merge branch 'cancel_cus' into rp_integration

jro1234 committed:
Save addition modelling output

jro1234 committed:
Merge branch 'cancel_cus' of https://github.com/jrossyra/adaptivemd into cancel_cus

jro1234 committed:
Cancel wait time updated

jro1234 committed:
Merge branch 'cancel_cus' into rp_integration

jro1234 committed:
Cleanup Execution scripts

jro1234 committed:
Small setup files cleanup

jro1234 committed:
Documentation cleanup

jro1234 committed:
docstring and method cleanup

jro1234 committed:
[project] initialize without configuration arg

jro1234 committed:
[configuration] eliminate broken syncvariables

jro1234 committed:
[configuration] fixed queue as list

jro1234 committed:
[tutorial] 1st tutorial with API updates

jro1234 committed:
Remove test working directories

jro1234 committed:
Removed simulation system changes

jro1234 commented:
@franknoe @thempel 
I apologize about having this PR incomplete before. I know it isn't good having so much committed here, if it is better to open a separate PR with these additional changes I would be happy to do that. 

All the changed content in the `adaptivemd.rp` module does not affect how the package is used as demonstrated in the example tutorials (i.e. the adaptivemdworker / cluster version of the code), hopefully that helps restrict the amount of review needed. 

I added some comments regarding the previous discussion, and will continue to update the documentation in existing code and work on the tests as I add new code for the workflow functionality in the near future.

franknoe commented:
Hi. Can you update the description of this PR, describing what are the features and tests introduced here? There seems to be a lot of commits after your first description, so I assume the original description of the PR is no longer valid.

For some commits such as "NTL9 in Charm22star" I don't understand if this is a general feature for this packages. Seems to relate to a specific application.

jro1234 commented:
@franknoe  Many of the commits are not specifically related to the functionality, I apologize about that. I tried to eliminate the changes to the repo from these commits by merging only changes to the code itself from my working branch into this PR branch. The file changes are not in the final diff, but these are still in the commit history unfortunately. 

I will go through the commits and do my best to list the commit messages and content changes to the cluster execution code, particularly for the older ones.

jro1234 commented:
I think the best way to organize is to give the module and a description of the changes, then I have listed the relevant commits below the description. If a series of commits was done on the issue, I only wrote the terminal ones with the others summarized by "..." to avoid cluttering up the space.  Many of the commits this spring are regarding the Radical Pilot module, I can add more information about these or any other questions about the code here.

[module] description
  - message from relevant commits

[analysis] RPC on MDFeaturizer with 'apply_feat_part' upgraded to allow general call signatures.
   - pyemma interface, error handling (August 2017)

[engine.openmm] shuffle file reading order to alleviate filesystem faults
   - Retry file reads . ... Retries Implemented (July 2017) 
   - fixed filereading ... minor changes (August 2017)

[unit test] simple test corrected to pass
   - [unittest] fixing  ... added directory check to test (August 2017)

[doc test] corrected to pass
   - [doctest] fixing (August 2017)

[project, mongodb] manipulate db location settings, access at project level
   - DB location (August 2017)
   - Update Project.py
   - Update mongodb.py (September 2017)

[project, resource, configuration, task] implement configuration-resource-task hierarchy for defining resource configurations and targeting tasks to resources. This was done without substantial changes to the API, a single line change when first starting up a project is all. Currently affects the Radical Pilot execution mechanism for most part, but will be used to expand the cluster execution functionality. 
   - Task-Resource-Configuration structure, unclean
   - [configuration] documentation
   - resource name in PythonTask
   - more CRT (August 2017)
   - condition upgraded .... [Configuration] set_current akin to old 'Resource' for adaptivemdworkers (September 2017)
   - Merge branch 'rp_integration' of .... (October 3, 2017)
   - Resource requirements passed from engine to task ... Fixed resource requirements typo (December 2017)
   - logging replace prints (March 2018)
   - [project] initialize without configuration arg  ....  [configuration] fixed queue as list (July 2018)

[project] blocking wait for multiple events (in workflow / task execution)
   - Take a list of conditions

[utils] parser for configuration format. currently only used to parse files that define a configuration object, but will be used for workflow parameters as well.
   - [config] parser moved to utils (October 2017)

[project, task] task path expansion handling and queueing
   - Changed queueing args and shared path expansion
   - Small changes, arg for always random sampling (September 2017)
   - [Project] queue list results in add list to bundle  (April 2018)

[mongodb] fixed circular loading from database when 2 objects each try to load the other. Holds single object reference, can expand this to list in case of circular reference with more than 2.
   - [mongodb] fixed 1-deep circular load call (RP disabled)  (October 2017)

[task]  environment management to load and deactivate conda or virtualenv based installations of the task environment (not necessarily adaptivemd package)
   - [task] provided env prepended to task descriptions ... deactivate environment if added (October 2017)
   - Added virtualenv to tasks (December 2017)
   - env loads first in pretask (June 2018)

[task] Trajectory Tasks inherit from PrePostTask. Moving all pre- and post- simulation operations to pretask and post task. (some of previous item crosses over here)
   - Trajectory Tasks inherit PrePostTask, add venv to prepost (December 2017)
   - mdconvert to pretask, changes to file (need to undo) (March 2018)

[bundle] Subclass hook to pick random item from bundle without loading all to memory if subclass has a fast method implemented. When picking from a StoredBundle with more than a few elements (i.e. sampling restart frames from stored trajectory files) , this develops into a significant bottleneck.  
   - Faster StoredBundle pick method (March 2018)

[mongodb] Bulk saving of list of storable objects by adding pymongo.insert_many call
   - [ObjectStore] save pathway added with insert_many capability 
   - [StoredBundle] handle group or item add  (April 2018)
   - Small Fixes . ...  [bulk_taskqueue] esolve merge conflicts  (April 2018)

jro1234 committed:
[rp] bool running property for Client

jro1234 committed:
[task] callback arg signature fixed

jro1234 committed:
[engine] use pretask to stage inputs

jro1234 committed:
Example chignolin system in Charmm22* ff

jro1234 committed:
Tutorial updates

jro1234 committed:
README updates

jro1234 committed:
README updates

jro1234 committed:
[project] queue function use *args

jro1234 committed:
[openmm] error if wrong input file extensions

jro1234 committed:
[tutorial] updates

jro1234 committed:
[project] first *args useage

jro1234 committed:
[util] get_logger to format & organize logging

jro1234 committed:
[tutorial] updates

jro1234 committed:
[travis] updated recipe, remove py3 for now

jro1234 committed:
[travis] comment out 'common_py_deps'

jro1234 committed:
[travis] comment out other 'common_py_deps'

jro1234 committed:
[travis] install conda-build

jro1234 committed:
[travis] small updates

jro1234 committed:
[rp] include rp in install

jro1234 committed:
[travis] radical.pilot added to meta.yaml

jro1234 committed:
[conda-recipe] trivial rearrange

jro1234 committed:
[conda-recipe] fixing

invalid-email-address committed:
removed RP from all test & setup for now

jro1234 committed:
[configuration] defaults for config fields

jro1234 committed:
add all rp

jro1234 committed:
[tests] use configuration objects

jro1234 committed:
[project] configuration from property

jro1234 committed:
fix deps and installs for test setup

jro1234 committed:
[tests] imports, others fixed

jro1234 committed:
[examples] smaller alanine test system

jro1234 committed:
Merge branch 'rp_integration' of https://github.com/jrossyra/adaptivemd into rp_integration

jro1234 committed:
[configuration] fixed modify field in init

jro1234 committed:
[configuration] see previous message

invalid-email-address committed:
[tests] worker tests, local travis passes

invalid-email-address committed:
[tests] fixed rp tests

jro1234 committed:
Update README.md

jro1234 committed:
Merge branch 'rp_integration' into py3_compat

jro1234 commented:
@franknoe @thempel I wanted to ping back about merging these changes. I've updated the README and tutorials, and made sure all the tests pass.

franknoe commented:
For me this is too much at once to digest. @thempel have you tried to use the package in this version?

thempel commented:
Hi @jrossyra , I'm happy to test it next week and will post my feedback here.

jro1234 commented:
@thempel Thanks, I'll keep my eye out for it. And I'll make sure to keep small and frequent contributions after we sort this one out.

thempel commented:
My assumption, that this PR ("py3 compat") is python 3 compatible, is wrong, isn't it? The installation `python setup.py install' failed with the following message:
```
SyntaxError: invalid syntax

  File "/storage/mi/thempel/anaconda3/envs/admd18/lib/python3.6/site-packages/AdaptiveMD-0.0.0-py3.6.egg/adaptivemd/rp/utils.py", line 258
    print traceback.format_exc()
                  ^
SyntaxError: invalid syntax
```
I guess radical pilot is the problem here. Would it run without rp with python 3? It would be great if all the dependencies would be mentioned in the installation instructions to be installed with conda...

jro1234 committed:
[adaptivemdworker] port, location opts for DB

thempel commented:
For the record: This branch was tested on our local HPC resource w/o radical pilot. @jrossyra made the DB-location adjustable so we can submit `adaptivemdworker` jobs to the cluster queue and control everything from our local machines. The basic example notebooks are running with a large test system and basic PyEMMA analysis with backbone torsion features. Further, unit tests are passing on travis-ci.

We note that further changes remain to be done. Since this PR is already extremely large, I will merge it now. Further feature requests and bugs can be handled using the github issue tracker.

Many thanks @jrossyra for your work!
